### PR TITLE
win_domain_user and win_domain_group: add created result boolean

### LINKED ISSUE #57547: [Feature Req] [win_domain_user] add a creation return value
##### SUMMARY

For the win_domain_user module, it would be nice to have a Boolean returned value called `created` as `changed` would not always been indicative of creation.

>  update_password: Note that `always` will always report an Ansible status of 'changed' because we cannot determine whether the new password differs from the old password.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
win_domain_user

##### ADDITIONAL INFORMATION
I think anyone that wants to manage service accounts or guest accounts where the password is always going to be set to a known value would have a need for being able to distinguish when a user is created vs when the account is first provisioned.

I'm currently just doing lab work, but I think the distinction would be nice and that there would be numerous use cases other then the example below.

```yaml
    - name: Generate kiosk accounts
      win_domain_user:
        name: kiosk1
        password: correcthorsebatterystaple
        state: present
        password_never_expires: yes
        user_cannot_change_password: yes
      register: guest_creation
    
    - name: add user to a firewall access policy
      uri:
        url: https://fw.example.com/api/users/untrusted_group/add/kiosk1?token={{ oauth_token }}
        method: POST
        status_code: 201
      when: guest_creation.created
```

### PR DESCRIPTION
**Pull request by ShachafGoldstein**
##### SUMMARY
Fixes #57547

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
win_domain_user.ps1
win_domain_group.ps1

##### ADDITIONAL INFORMATION
Added to both to keep them alike
Added changes to test to validate correct value on different cases

### CODE REVIEW

**Comment by ShachafGoldstein** [comment]
@jborean93 @nitzmahone - Please review when you have the time

**Comment by jhawkesworth** [comment]
Would want to be sure that the changes proposed here would't be affected by the change proposed here: https://github.com/ansible/ansible/issues/58246#issuecomment-505613127

**Comment by ShachafGoldstein** [comment]
> Would want to be sure that the changes proposed here would't be affected by the change proposed here: [#58246 (comment)](https://github.com/ansible/ansible/issues/58246#issuecomment-505613127)

No clash, maybe merge conflicts but they are separate and not dependent

**Review by jborean93 — CHANGES_REQUESTED** [review]
Just need a rebase (sorry about that) and update the actual documentation to show this new return value.

File: lib/ansible/modules/windows/win_domain_group.py [hunk 1]
```diff
@@ -234,4 +234,9 @@
   returned: group exists
   type: str
   sample: S-1-5-21-2171456218-3732823212-122182344-1189
+created:
+  description: Whether a group was created
+  returned: when a new group was created
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_domain_group.py]: just a nitpick, this should be always as we return `$false` in the case of a group not being created. Same as the `win_domain_user` entry.

File: lib/ansible/modules/windows/win_domain_user.ps1 [hunk 2]
```diff
@@ -112,6 +114,31 @@ If ($state -eq 'present') {
         $result.changed = $true
     }
 
+    # Configure password policies
+    If (($null -ne $password_never_expires) -and ($password_never_expires -ne $user_obj.PasswordNeverExpires)) {
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_domain_user.ps1]: What's happening here? Seems like it should be in a separate PR or just add a changelog entry.

**Review by jborean93 — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_domain_user.ps1 (hunk 2 above)
Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_domain_user.ps1]: HAd some issues with this after you merged the removal of the try from this. - the indentations were off and the if block was not closed - ill give it a second look against the devel current file

Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_domain_user.ps1]: Fixed

File: lib/ansible/modules/windows/win_domain_user.ps1 [hunk 3]
```diff
@@ -96,6 +97,7 @@ If ($state -eq 'present') {
             New-ADUser -Name $username -WhatIf:$check_mode @extra_args
         }
         $new_user = $true
+		$result.created = $true
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_domain_user.ps1]: Slight indentation problem.

File: lib/ansible/modules/windows/win_domain_user.ps1 (hunk 2 above)
Comment (jborean93) [reply lib/ansible/modules/windows/win_domain_user.ps1]: Ahh cool so it was just a rebase gone wrong and not a problem with my PR?

Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_domain_user.ps1]: lol yeah

File: lib/ansible/modules/windows/win_domain_user.ps1 (hunk 3 above)
Comment (ShachafGoldstein) [reply lib/ansible/modules/windows/win_domain_user.ps1]: oops

**Review by jborean93 — APPROVED** [review]
